### PR TITLE
Add lint option to Makefile and fix lint issues

### PR DESCRIPTION
**Pull request by Rulox**
### Proposed changes
Add a Makefile option to run go linter in dev.

### Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-asg-sync/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

@pleshakov Do you think it makes sense to `exit(1)` if session.NewSession() returns an err? (please see the changes)

### CODE REVIEW

File: cmd/sync/main.go [hunk 1]
```diff
@@ -145,7 +145,12 @@ func main() {
 func createAWSClient(region string) *AWSClient {
 	httpClient := &http.Client{Timeout: connTimeoutInSecs * time.Second}
 	cfg := &aws.Config{Region: aws.String(region), HTTPClient: httpClient}
-	session := session.New(cfg)
+	session, err := session.NewSession(cfg)
```
Comment (pleshakov) [inline cmd/sync/main.go]: we should propagate this error to the main func and handle it as in https://github.com/nginxinc/nginx-asg-sync/blob/55f3dbabfaee6a4d01a0a1e9cb6f15710ee758c8/cmd/sync/main.go#L56
It makes sense to make this change in a separate commit . seems like a small bug

Comment (Rulox) [reply cmd/sync/main.go]: I can make the change here if needed/

**Comment by Rulox** [comment]
@pleshakov I have added another commit to handle the error in the `main` goroutine. Let me know your thoughts. Thanks
